spec/jobs/emit_web_hook_event_spec.rb

```diff
@@ -125,6 +125,33 @@
     end
   end
 
+  context 'with tag filters' do
+    let(:tag) { Fabricate(:tag) }
+    let(:topic) { Fabricate(:topic, tags: [tag]) }
+    let(:topic_hook) { Fabricate(:topic_web_hook, tags: [tag]) }
+
+    it "doesn't emit when event is not related with defined tags" do
+      subject.execute(
+        web_hook_id: topic_hook.id,
+        event_type: 'topic',
+        payload: { test: "some payload" }.to_json
+      )
+    end
+
+    it 'emit when event is related with defined tags' do
+      stub_request(:post, "https://meta.discourse.org/webhook_listener")
+        .with(body: "{\"topic\":{\"test\":\"some payload\"}}")
+        .to_return(body: 'OK', status: 200)
+
+      subject.execute(
+        web_hook_id: topic_hook.id,
+        event_type: 'topic',
+        tag_ids: topic.tags.pluck(:id),
+        payload: { test: "some payload" }.to_json
+      )
```

Comment:
we should add one more test here that confirms it does not raise any web hook when the tag does not match.